ext/cl_khr_tensor.asciidoc

```diff
@@ -0,0 +1,547 @@
+
+// TODO: add clSetTensorObjectDestructorCallback?
+
+To return information about a tensor object, call the function
+
+[source,c]
+----
+cl_int clGetTensorInfo(
+  cl_tensor tensor,
+  cl_tensor_info param_name,
+  size_t param_value_size,
+  void* param_value,
+  size_t* param_value_size_ret);
+----
+
+* _tensor_ specifies the tensor object being queried.
+
+* _param_name_ specifies the information to query. The list of
+  supported param_name types and the information returned in
+  _param_value_ by clGetTensorInfo is described in the <<Tensor Object
+  Queries>> table.
+
+* _param_value_ is a pointer to memory where the appropriate result
+  being queried is returned. If _param_value_ is NULL, it is ignored.
+
+* _param_value_size_ is used to specify the size in bytes of memory
+  pointed to by _param_value_. This size must be ≥ size of return type
+  as described in the <<Tensor Object Queries>> table.
+
+* _param_value_size_ret_ returns the actual size in bytes of data
+  being queried by _param_name_. If _param_value_size_ret_ is NULL, it is
+  ignored.
+
+*clGetTensorInfo* returns CL_SUCCESS if the function is executed
+ succesfully. Otherwise, it returns one of the following errors:
+
+* CL_INVALID_TENSOR if _tensor_ is not a valid tensor object.
+
+[#Tensor Object Quaries]
+.List of supported param_names by clGetTensorInfo
+[cols="2,1,2",stripes=odd]
+|===
+| CL_TENSOR_RANK  | size_t         | Return the tensor rank.
+| CL_TENSOR_SHAPE | size_t[]       | Return the tensor shape.
+| CL_TENSOR_DTYPE | cl_tensor_type | Return the tensor data type.
+
+| CL_TENSOR_COMMAND_BUFFER_TEMPORARY | cl_bool | Return true if the
+tensor is temporary tensor for command buffers.
+
+| CL_TENSOR_BOUND_TO_BUFFER | cl_bool | Return true if the tensor is
+bound to a buffer. If CL_TENSOR_COMMAND_BUFFER_TEMPORARY is true, then
+CL_TENSOR_BOUND_TO_BUFFER must return false.
+
+| CL_TENSOR_BUFFER | cl_mem a| If CL_TENSOR_BOUND_TO_BUFFER is true,
+return the buffer object the tensor is bound to. Otherwise,
+clGetTensorInfo call returns:
+
+* CL_INVALID_MEM_OBJECT if the tensor is not bound to a buffer object.
+
+* CL_INVALID_PROPERTY otherwise.
+
+| CL_TENSOR_CONTEXT | cl_context | Return the context specified when
+  the tensor object is created.
+
+| CL_TENSOR_REFERENCE_COUNT | cl_uint | Return the tensor reference
+count.
+|===
+
+To read from a tensor to host memory / buffer object or to write to a
+tensor object from host memory / buffer object call one of the functions.
+
+[source,c]
+----
+cl_int clEnqueueReadTensor(
+  cl_command_queue command_queue,
+  cl_tensor tensor,
+  cl_bool blocking_command,
+  cl_mem buffer,
+  void* host_ptr,
+  cl_uint num_events_in_wait_list,
+  const cl_event* event_wait_list,
+  cl_event* event);
+----
+
+[source,c]
+----
+cl_int clEnqueueWriteTensor(
+  cl_command_queue command_queue,
+  cl_tensor tensor,
+  cl_bool blocking_command,
+  cl_mem buffer,
+  void* host_ptr,
```

Comment:
This should be `const` for the write API.  Not sure if other changes are needed here too (possible copy-paste?).

```suggestion
  const void* host_ptr,
```